includes/internal/installer/migrations/class-student-progress-migration.php

```diff
@@ -0,0 +1,491 @@
+			$post_meta = $wpdb->get_results( $post_meta_query );
+		}
+
+		// Map different meta keys to the comment ID.
+		$mapped_meta = [];
+
+		// Map comment meta to the comment ID.
+		foreach ( $progress_comment_meta as $meta ) {
+			// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+			$mapped_meta[ $meta->comment_id ][ $meta->meta_key ] = $meta->meta_value;
+		}
+
+		// Map post meta to the comment ID.
+		foreach ( $post_meta as $meta ) {
+			$comment_id = $post_ids[ $meta->post_id ];
+
+			if ( '_wp_trash_meta_comments_status' === $meta->meta_key ) {
+				$comment_statuses = maybe_unserialize( $meta->meta_value );
+				if ( ! is_array( $comment_statuses ) ) {
+					continue;
+				}
+
+				foreach ( $comment_statuses as $comment_id => $comment_status ) {
+					$mapped_meta[ $comment_id ]['status'] = $comment_status;
+				}
+			} else {
+				// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+				$mapped_meta[ $comment_id ][ $meta->meta_key ] = $meta->meta_value;
+			}
+		}
+
+		$last_comment_id = end( $comment_ids );
+
+		return array( $progress_comments, $mapped_meta, $last_comment_id );
+	}
+
+
+	/**
+	 * Prepare the course progress data to be inserted.
+	 *
+	 * @param array $progress_comments The comments to migrate.
+	 * @param array $mapped_meta The comment meta to migrate.
+	 */
+	private function prepare_progress_to_insert( $progress_comments, $mapped_meta ): void {
+		$this->progress_inserts = array();
+		foreach ( $progress_comments as $progress_comment ) {
+			$meta = isset( $mapped_meta[ $progress_comment->comment_ID ] )
+				? $mapped_meta[ $progress_comment->comment_ID ]
+				: array();
+
+			// Process comments for trashed posts.
+			if ( 'post-trashed' === $progress_comment->comment_approved ) {
+				$progress_comment->comment_approved = isset( $meta['status'] )
+					? $meta['status']
+					: 'in-progress';
+			}
+
+			switch ( $progress_comment->comment_type ) {
+				case 'sensei_course_status':
+					$this->prepare_course_progress_to_insert( $progress_comment, $meta );
+					break;
+				case 'sensei_lesson_status':
+					$this->prepare_lesson_progress_to_insert( $progress_comment, $meta );
+					break;
+				default:
+					$this->errors[] = sprintf(
+						/* translators: %s: comment type */
+						__( 'Unknown comment (id %1$d) type: %2$s', 'sensei-lms' ),
+						$progress_comment->comment_ID,
+						$progress_comment->comment_type
+					);
+
+			}
+		}
+	}
+
+	/**
+	 * Prepare the course progress data to be inserted.
+	 *
+	 * @param object $comment The comment to migrate.
+	 * @param array  $meta The comment meta to migrate.
+	 */
+	private function prepare_course_progress_to_insert( $comment, $meta ): void {
+		$course_status = 'in-progress';
+		if ( Course_Progress::STATUS_COMPLETE === $comment->comment_approved ) {
+			$course_status = 'complete';
+		}
+
+		if ( Course_Progress::STATUS_COMPLETE === $comment->comment_approved ) {
+			$completed_at = $comment->comment_date;
```

Comment:
Did we plan to store dates in GMT for the new tables? If so, I don't think the `comment_date` is in GMT and might need some conversion. Maybe we can use `comment_date_gmt` instead?

Reply:
@m1r0 added usage of GMT date and conversion for the `start` date.